### PR TITLE
bug: [PAGOPA-1377] out of memory error

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- removed EntityGraph to allow offset and limit

#### Motivation and Context

- out of memory error for in memory pagination

#### How Has This Been Tested?

Local test

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/debtposition/repository/specification/PaymentOptionByAttribute.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                        cb.between(root.get(IUV_FIELD), segregationCode, getSegregationCodeEnd(segregationCode)));
+            }
+        } else {
+            segregationCodesPredicate = cb.isTrue(cb.literal(true));
+        }
+
+        return cb.and(paymentPositionIdPredicate, dueDatePredicate, segregationCodesPredicate);
+    }
+
+    private String getSegregationCodeEnd(String segregationCode) {
```
Comment (AngeloCaporaso) [inline src/main/java/it/gov/pagopa/debtposition/repository/specification/PaymentOptionByAttribute.java:63]: this is used also in `PaymentPositionByOptionsAttribute`, maybe this method `getSegregationCodeEnd` could be pushed up in `Util` class.

File: src/main/resources/db/migration/V010__ADD_PP_INDEXES.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CREATE INDEX idx_company_name ON payment_position (company_name);
```
Comment (AngeloCaporaso) [inline src/main/resources/db/migration/V010__ADD_PP_INDEXES.sql:1]: `company_name` is rarely used, consider whether to keep it and to change the default sorting logic (now is based on `company_name`)
cc: @aferracci @pagopa/pagopa-team-core

**Review by AngeloCaporaso — APPROVED** [review]
There is some suggestion in the review comments, anyway, good work 🥇

File: src/main/java/it/gov/pagopa/debtposition/repository/specification/PaymentOptionByAttribute.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/debtposition/repository/specification/PaymentOptionByAttribute.java:63]: done

**Pull request merged**